self-core-impl/src/main/java/com/selfxdsd/core/JsonResources.java

```diff
@@ -214,15 +214,13 @@ private HttpRequest request(
             final HttpRequest.BodyPublisher body
         ) {
             final HttpRequest request;
-            if(this.accessToken != null && !this.accessToken.isEmpty()) {
+            if(this.accessToken != null) {
                 request = HttpRequest.newBuilder()
                     .uri(uri)
                     .method(method, body)
                     .header("Content-Type", "application/json")
-                    .header(
-                        "Authorization",
-                        "token " + this.accessToken
-                    ).build();
+                    .header(this.accessToken.header(), this.accessToken.value())
```

Comment:
@criske In the case of Github, you have to preppend the ``token `` word in front of the actual token's value. Maybe you can modify the ``value()`` method to return ``"token " + value``?

Reply:
@amihaiemil It's prepended in GithubAccessToken constructor.